cmd/sriovdp/manager.go

```diff
@@ -208,28 +211,14 @@ func (rm *resourceManager) discoverHostDevices() error {
 			}
 			glog.Infof("discoverDevices(): device found: %-12s\t%-12s\t%-20s\t%-40s", device.Address, device.Class.ID, vendorName, productName)
 
-			// exclude device in-use in host
-			if isDefaultRoute, _ := hasDefaultRoute(device.Address); !isDefaultRoute {
-
-				aPF := utils.IsSriovPF(device.Address)
-				aVF := utils.IsSriovVF(device.Address)
-
-				if aPF || !aVF {
-					// add to linkWatchList
-					rm.addToLinkWatchList(device.Address)
-				}
-
-				if aPF && utils.SriovConfigured(device.Address) {
-					// do not add this device in net device list
-					continue
-				}
-
-				if newDevice, err := resources.NewPciNetDevice(device, rm.rFactory); err == nil {
-					rm.netDeviceList = append(rm.netDeviceList, newDevice)
-				} else {
-					glog.Errorf("discoverDevices() error adding new device: %q", err)
-				}
+			// Network devices
+			if devClass == netClass {
+				rm.netDevHandler(device)
+			}
 
+			// Processing accelerators
+			if devClass == accClass {
```

Comment:
It seems a little strange to have two totally separate if statements where only one of them can be true...normally I'd expect to see an "else if" here.